### ISSUE TITLE
[QUESTION] 

Just to be sure, no impact due to CVE-2023-32784?

### ISSUE THREAD

**Issue by Friedrieck** [PhilippC/keepass2android#2340]
**What version of Keepass2Android are you using?**
Please follow these steps to find your app version:
1. Click the **⁝** icon in the top right corner
2. Select **Settings**
3. Click **About**
4. Find the "Version" information and provide it here:

1.09e-r7

**Comment by ikanakova** [comment]
I think this vulnerability is exclusive to Windows (see [CVE-2023-32784](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-32784)) and moreover for example KeePassXC are not impacted (see [KeePass 2.X Master Password Dumper](https://github.com/vdohney/keepass-password-dumper)).

**Comment by LunaticoCR** [comment]
Good to know! 😃

**Comment by felix32257** [comment]
Mmhh... Wouldn't call that "knowing" since the developer hasn't said anything about this. To my understanding this is especially bad under windows because of .NET used by Keepass2 but Linux and Mac are possibly affected as well. It was even tested under Debian... Until the developer responds or somebody who knows the code better, this question remains open. If you want to be sure you must assume the worst case, clean up and wait for an update. Maybe, just maybe, encrypted sandboxes like Samsung's "secure folder" might help with this. But maybe not at all either.

**Comment by ikanakova** [comment]
@felix32257 You're right. As I understand it, the vulnerability is exclusive to .NET. Official KeePass 2.X is only for Windows, for Linux or MacOS you can use either a Windows emulator or unofficial KeePass Packages through the use of Mono, that uses .NET. So if Keepass2Android uses also .NET, it may probably also have the CVE-2023-32784 vulnerability, hopefully @PhilippC  will give us an answer.

**Comment by the-jade-rabbitt** [comment]
I came here with the same concern!  While we wait for an answer, searching for .net in the source code: docs\Keepass2Android-Apk.md states that keepass2android is using Mono, an open source version of .net.

@felix32257 @ikanakova what can be said about the security of this version of .net?

@PhilippC it would be good to get a resolution on this - I promise to become a paying sponsor if you can!



**Keepass2Android-Apk.md:**
```
Keepass2Android's apk is pretty big, e.g. when comparing to Keepassdroid. 
The main difference is that Keepass2Android is built on Mono for Android. 
Mono is an open-source implementation of the Microsoft .Net Framework (installed on pretty much every Windows PC). 
On Windows, the .net framework requires several hundred MB (but only once, not for every application). On Android devices,
Mono is not installed globally. Instead, it is packaged into every app. 
The more features from Mono are required, the bigger the package becomes.

Here's a list of what is contained in the Keepass2Android 0.9.1 application package:

Mono for Android		
			.net dlls			5.0 MB
			Runtime				2.5 MB				
			Google libraries		0.8 MB				
			(for Drive support)

Resources		Strings, Icons..		2.1 MB				
Password Font						0.2 MB				
Java Code		including Dropbox 		1.1 MB				
				GDrive, SkyDrive
				libraries
							
Keepass library						0.2 MB				
Keepass2Android Code					0.3 MB				
Java/Mono bindings					0.5 MB				
							
rest							0.3 MB		

TOTAL							13 MB
```

**Comment by the-jade-rabbitt** [comment]
Also see this:
https://github.com/keepassxreboot/keepassxc/discussions/9433

> KeePass 2.X uses a custom-developed text box for password entry, SecureTextBoxEx. 
> This text box is not only used for the master password entry, but in other places in KeePass as well, 
> like password edit boxes (so the attack can also be used to recover their contents).
> 
> The flaw exploited here is that for every character typed, a leftover string is created in memory. 
> Because of how .NET works, it is nearly impossible to get rid of it once it gets created. 
> For example, when "Password" is typed, it will result in these leftover strings: 
> •a, ••s, •••s, ••••w, •••••o, ••••••r, •••••••d. 
> The POC application searches the dump for these patterns and offers a likely password character for each position in the password.

A search for SecureTextBoxEx in keepass2android source comes up empty, so hopefully the password entry uses an native android API for this ...

**Comment by felix32257** [comment]
I don't know. I don't know whether or not the problematic function was used here as well (because why would you replace it?). Maybe. Maybe not.

I also can only assume that Mono has the same problem because I think the successful testing of this under Debian would have been with Mono.

Overall it's not exactely productive what we are talking here because basically it's all only assumption... Only dev or someone who doesn't just speculate can tell better. :) until then, personally I will assume the worst, i.e. vulnerable.
